packages/auth-helpers/src/lib/craft-auth-sig.ts

```diff
@@ -0,0 +1,50 @@
+import { AuthSig, SignerLike } from '@lit-protocol/types';
+import { ethers } from 'ethers';
+
+/**
+ * Generate an AuthSig object using the signer.
+ *
+ * For more context:
+ * We are only using authSig to generate session sigs. In a newer version, we will stop accepting
+ * authSig all together from the node and will only accept session sigs.
+ *
+ * @param signer the signer must have a "signMessage" method
+ * @param toSign - the message to sign
+ * @param address - (optional) the address of the signer
+ * @returns
+ */
+export const generateAuthSig = async ({
+  signer,
+  toSign,
+  address,
+  algo,
+}: {
+  signer: ethers.Wallet | ethers.Signer | SignerLike;
+  toSign: string;
+  address?: string;
+  algo?: 'LIT_BLS' | 'ed25519';
+}): Promise<AuthSig> => {
+  if (!signer?.signMessage) {
+    throw new Error('signer does not have a signMessage method');
+  }
+
+  const signature = await signer.signMessage(toSign);
+
+  // If address is not provided, derive it from the signer
+  if (!address) {
```

Comment:
The address should be always the same as the signer?

Reply:
yeah i think it has to come from the signer?  like what if they pass a wrong, non matching address?  is there a case for this?  or do we think they'll always be using the address from the signer, in which case, we could remove that param?

Reply:
Yea, it handles the case where the signer being used is not from `ethers` so they might have to provide their own address.